core/src/main/java/hudson/util/ProcessKillingVeto.java

```diff
@@ -47,7 +49,7 @@
  * 
  * @author <a href="mailto:daniel.weber.dev@gmail.com">Daniel Weber</a>
  */
-public abstract class ProcessKillingVeto implements ExtensionPoint {
+public abstract class ProcessKillingVeto implements ExtensionPoint, Serializable {
```

Comment:
FWIW, I was able to test this out this morning on an instance that has the current released version of the MSBuild plugin and it worked properly.  To my knowledge, MSBuild plugin is the only known implementation of this ExtensionPoint so far (or at least the only one in Jenkins GitHub).

I ran a job both on the slave and on master and each time the mspdbsrv process survives the job completion and I see a message in the log recorder that killing that process id has been veto'd.